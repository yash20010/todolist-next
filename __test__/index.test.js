import React from 'react'
import ReactDOM from 'react-dom'
import App from '../pages/index.js'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App tasks={[]} />, div)
})

it('renders the input form, add button, checkbox, todo text, edit button and delete button correctly', () => {
  const { getByTestId } = render(
    <App
      tasks={[
        {
          _id: 1,
          task: 'hello',
          completed: false,
          createdAt: new Date(),
        },
      ]}
    />
  )
  expect(getByTestId('input')).toBeInTheDocument()
  expect(getByTestId('button')).toBeInTheDocument()
  expect(getByTestId('checkbox')).toBeInTheDocument()
  expect(getByTestId('task_text1')).toBeInTheDocument()
  expect(getByTestId('edit-btn')).toBeInTheDocument()
  expect(getByTestId('delete-btn')).toBeInTheDocument()
})

it('matches snapshot', () => {
  const tree = renderer.create(<App tasks={[]} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('matches snapshot with todos', () => {
  const tasks = [
    {
      _id: 1,
      task: 'hello',
      completed: false,
      createdAt: new Date(),
    },
    {
      _id: 2,
      task: 'world',
      completed: false,
      createdAt: new Date(),
    },
  ]
  const tree = renderer.create(<App tasks={tasks} />).toJSON()
  expect(tree).toMatchSnapshot()
})
