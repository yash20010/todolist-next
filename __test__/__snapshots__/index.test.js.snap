// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main
  className="main"
>
  <h1
    className="heading"
  >
    TO-DO
  </h1>
  <div
    className="container"
  >
    <form
      className="form_container"
      onSubmit={[Function]}
    >
      <input
        className="input"
        data-testid="input"
        onChange={[Function]}
        placeholder="Task to be done..."
        type="text"
        value=""
      />
      <button
        className="submit_btn"
        data-testid="button"
        disabled={true}
        type="submit"
      >
        Add
      </button>
    </form>
    <h2
      className="no_tasks"
    >
      No tasks
    </h2>
  </div>
</main>
`;

exports[`matches snapshot with todos 1`] = `
<main
  className="main"
>
  <h1
    className="heading"
  >
    TO-DO
  </h1>
  <div
    className="container"
  >
    <form
      className="form_container"
      onSubmit={[Function]}
    >
      <input
        className="input"
        data-testid="input"
        onChange={[Function]}
        placeholder="Task to be done..."
        type="text"
        value=""
      />
      <button
        className="submit_btn"
        data-testid="button"
        disabled={true}
        type="submit"
      >
        Add
      </button>
    </form>
    <div
      className="task_container"
      data-testid="task-container"
    >
      <input
        checked={false}
        className="check_box"
        data-testid="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <p
        className="task_text"
        data-testid="task_text1"
      >
        hello
      </p>
      <button
        className="edit_task"
        data-testid="edit-btn"
        onClick={[Function]}
      >
        ✎
      </button>
      <button
        className="remove_task"
        data-testid="delete-btn"
        onClick={[Function]}
      >
        ✖
      </button>
    </div>
    <div
      className="task_container"
      data-testid="task-container"
    >
      <input
        checked={false}
        className="check_box"
        data-testid="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <p
        className="task_text"
        data-testid="task_text2"
      >
        world
      </p>
      <button
        className="edit_task"
        data-testid="edit-btn"
        onClick={[Function]}
      >
        ✎
      </button>
      <button
        className="remove_task"
        data-testid="delete-btn"
        onClick={[Function]}
      >
        ✖
      </button>
    </div>
  </div>
</main>
`;
